src/shaders/conjurer_common.frag

```diff
@@ -43,33 +49,79 @@ float plot(vec2 st, float pct) {
         smoothstep(pct, pct + 0.02, st.y);
 }
 
+// ========== COORDINATE PROJECTIONS ==========
+
+// We make use of a number of coordinate systems. Importantly, by default shaders are provided with
+// a "varying" called v_uv which is a vec2 in what we call "canopy coordinates".
+
+// ---------- CANOPY COORDINATES ----------
+
+// Think of these coordinates as: x is the angle around the center of the canopy, and y is the
+// distance away from the apex. It is very similar to polar coordinates (theta, radius), but with a
+// different radius. In the case of canopy coordinates, a "radius" of 0.0 does not mean the exact
+// center of the circle, but rather means the edge of the apex. Or, the distance to the first led.
+
+// Canopy coordinates are normalized:
+//   x goes from 0.0 to 1.0, which describes the angle around the center from 0 to 2pi
+//   y goes from 0.0 to 1.0, which describes the distance from the apex to the edge of the canopy
+
+// Cannot be coordinate can be useful to write patterns so that you can described behavior with the
+// boundaries of the canopy in mind. However, the coordinate space being so specialized can be
+// limiting. Use the functions below to convert to other coordinate systems whenever desired.
+
+// ---------- CARTESIAN COORDINATES ----------
+
+// Cartesian coordinates mean the usual x,y coordinates, where:
+//  (0, 0) is at the center of the canopy
+//  (1.0, 1.0) is at the top right corner of the canopy
+//  (-1.0, -1.0) is at the bottom left corner of the canopy
+
+// ---------- HALF CARTESIAN COORDINATES ----------
+
+// For lack of a better name, "half cartesian coordinates" is what we will call cartesian coordinates but with
+// the axes going from -0.5 to 0.5, so:
+//  (0, 0) is at the center of the canopy
+//  (0.5, 0.5) is at the top right corner of the canopy
+//  (-0.5, -0.5) is at the bottom left corner of the canopy
+
+// ---------- POLAR COORDINATES ----------
+
+// Polar coordinates are the usual theta, radius coordinates, where:
+//  theta goes from 0.0 to 1.0, which describes the angle around the center from 0 to 2pi
+//  radius goes from 0.0 to 1.0, which describes the distance from the center to the edge of the canopy
+
+// ---------- ---------- ---------- ----------
+
 // Converts canopy coordinates to cartesian coordinates
-// Canopy coordinates mean:
-//  x: 0.0 means the location of the first led on the first strip
-//  x: 1.0 means the location of the first led on the last strip
-//  y: 0.0 means the location of the last led on the first strip
-//  y: 1.0 means the location of the last led on the last strip
-// Cartesian coordinates mean the usual x,y coordinates, where (0, 0) is at the center of the canopy
-// and (0.5, 0.5) is at the top right corner of the canopy.
 vec2 canopyToCartesianProjection(vec2 _st) {
-    float theta = (_st.x - 0.5) * 2.0 * PI;
-    // TODO: double check these numbers
+    float theta = _st.x * 2.0 * PI;
```

Comment:
@justynaipham also, I kept the change below but reverted this change. I think that by subtracting 0.5 from _st.x, we ended up rotating the coordinates around by pi (since x is essentially theta in canopy coordinates), which made y go from negative to positive from top to bottom, respectively. I know that some canvases follow that convention, but I think it's more common for positive to be on top. Anyways, Hopefully we are in a good place now!